ingestion/src/metadata/ingestion/source/mlmodel/sagemaker.py

```diff
@@ -0,0 +1,176 @@
+                    "arn": model["ModelArn"],
+                    "creation_timestamp": model["CreationTime"].isoformat(),
+                }
+            )
+
+    def _get_algorithm(self) -> str:  # pylint: disable=arguments-differ
+        logger.info(
+            "Setting algorithm to default value of `mlmodel` for SageMaker Model"
+        )
+        return "mlmodel"
+
+    def yield_mlmodel(  # pylint: disable=arguments-differ
+        self, model: SageMakerModel
+    ) -> Iterable[CreateMlModelRequest]:
+        """
+        Prepare the Request model
+        """
+        self.status.scanned(model.name)
+
+        yield CreateMlModelRequest(
+            name=model.name,
+            algorithm=self._get_algorithm(),  # Setting this to a constant
+            mlStore=self._get_ml_store(model.name),
+            service=EntityReference(
+                id=self.context.mlmodel_service.id, type="mlmodelService"
+            ),
+        )
+
+    def _get_ml_store(  # pylint: disable=arguments-differ
+        self,
+        model_name: str,
+    ) -> Optional[MlStore]:
+        """
+        Get the Ml Store for the model
+        """
+        try:
+            model_info = self.sagemaker.describe_model(ModelName=model_name)
+            return MlStore(imageRepository=model_info["PrimaryContainer"]["Image"])
+        except ValidationError as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Validation error adding the MlModel store from model description: {model_name} - {err}"
+            )
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Wild error adding the MlModel store from model description: {model_name} - {err}"
+            )
+        return None
+
+    def _get_tags(self, model_arn: str) -> Optional[List[TagLabel]]:
+        try:
+            tags = self.sagemaker.list_tags(ResourceArn=model_arn)["Tags"]
+            return [
+                TagLabel(
+                    tagFQN=tag["Key"],
+                    description=tag["Value"],
+                    source="Tag",
+                    labelType="Propagated",
+                    state="Confirmed",
+                )
+                for tag in tags
+            ]
+        except ValidationError as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Validation error adding TagLabel from model tags: {model_arn} - {err}"
+            )
+        except Exception as err:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Wild error adding TagLabel from model tags: {model_arn} - {err}"
+            )
+        return None
+
+    def _get_hyper_params(self, *args, **kwargs) -> Optional[List[MlHyperParameter]]:
+        pass
+
+    def _get_ml_features(self, *args, **kwargs) -> Optional[List[MlFeature]]:
+        pass
```

Comment:
Oh, is there no way to pick up the features from the API? It would be one of the most interesting pieces IMO to explore similar models and possibly reuse implementations across DS teams

Reply:
I tried to search for it but couldn't find anything. I don't think SageMaker stores info about the data features on which the model was trained on.Please let me know if you find anything, I'll be happy to add it.